lib/doorkeeper/grant_flow.rb

```diff
@@ -0,0 +1,54 @@
+require 'doorkeeper/grant_flow/flow'
+
+module Doorkeeper
+  module GrantFlow
+    mattr_accessor :registered_flows
+    self.registered_flows = {}
+
+    module_function
```

Comment:
> Can we make this a singleton class, rather than a module?

I see doorkeeper uses this pattern elsewhere, I don't like that. As this is new code, hopefully we can refactor it into classes.